### ISSUE TITLE
Setting gridOptions for grouping, without data, gives an error

### ISSUE THREAD

**Issue by joostvunderink** [ceolter/angular-grid#133]
Setting $scope.gridOptions with

```rowsAlreadyGrouped: true,
   rowData: null```

Leads to an error `Cannot read property 'forEach' of undefined` in `inMemoryRowController.js`.

This occurs when you don't have your rowData right away, but are getting it asynchronously.

Workaround is to not set `rowsAlreadyGrouped` initially, but set it together with `rowData` once you have retrieved your data.

Code here: http://garion.org/angular-grid/fileBrowser.html
With workaround: http://garion.org/angular-grid/fileBrowser2.html

Closer description of the bug is in the comments in `fileBrowser.js`.

**Comment by ceolter** [comment]
thanks for raising

**Comment by ceolter** [comment]
fixed in 1.7.0
